components/mediation-connector/org.wso2.carbon.mediation.connector.as4/src/main/java/org/wso2/carbon/mediation/connector/AS4ErrorHandler.java

```diff
@@ -0,0 +1,94 @@
+/*
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.mediation.connector;
+
+import org.apache.axiom.om.OMAbstractFactory;
+import org.apache.axiom.om.OMNode;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.mediation.connector.exceptions.AS4ErrorMapper;
+import org.wso2.carbon.mediation.connector.exceptions.AS4Exception;
+import org.wso2.carbon.mediation.connector.message.beans.MessageInfo;
+import org.wso2.carbon.mediation.connector.message.beans.Messaging;
+import org.wso2.carbon.mediation.connector.message.beans.SignalMessage;
+import org.wso2.carbon.mediation.connector.message.beans.Error;
+import org.wso2.carbon.mediation.connector.message.util.AS4Utils;
+import org.wso2.carbon.mediation.connector.message.util.MessageIdGenerator;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Marshaller;
+import java.util.Date;
+
+/**
+ * Handle AS4 Error messages
+ */
+
+public class AS4ErrorHandler {
+
+    private static final Log log = LogFactory.getLog(AS4ErrorHandler.class);
+
+    /**
+     * Generate AS4 error messages and set to {@link MessageContext}
+     * @param messageContext {@link MessageContext} object
+     * @param e {@link AS4Exception} object
+     */
+    public static void generateErrorMessage(MessageContext messageContext, AS4Exception e) {
```

Comment:
Fixed in https://github.com/wso2/carbon-mediation/pull/842/commits/28dc8826d26664836b4c45b5207d5ddbfde0fb41